cleanlab/rank.py

```diff
@@ -152,14 +164,320 @@ def get_normalized_margin_for_each_label(
 
     Returns
     -------
-      normalized_margin : np.array (float)
-        Return a score (between 0 and 1) for each example of its likelihood of
-        being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
-        normalized_margin = prob_label - max_prob_not_label"""
+    normalized_margin : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
+      normalized_margin = prob_label - max_prob_not_label"""
 
     self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
     max_prob_not_label = np.array(
         [max(np.delete(pred_probs[i], l, -1)) for i, l in enumerate(labels)]
     )
-    label_quality_scores = (self_confidence - max_prob_not_label + 1) / 2
+    normalized_margin = (self_confidence - max_prob_not_label + 1) / 2
+    return normalized_margin
+
+
+def get_confidence_weighted_entropy_for_each_label(
+    labels: np.array, pred_probs: np.array
+) -> np.array:
+    """Returns the normalized entropy divided by "self-confidence".
+
+    Score is between 0 and 1.
+    1 - clean label (not an error).
+    0 - dirty label (label error).
+
+    Parameters
+    ----------
+    labels : np.array
+      A discrete vector of noisy labels, i.e. some labels may be erroneous.
+      *Format requirements*: for dataset with K classes, labels must be in {0,1,...,K-1}.
+
+    pred_probs : np.array (shape (N, K))
+      P(label=k|x) is a matrix with K model-predicted probabilities.
+      Each row of this matrix corresponds to an example x and contains the model-predicted
+      probabilities that x belongs to each possible class.
+      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
+      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+
```

Comment:
We probably don't want to copy/paste argument definitions for each individual scoring function, as then updating them will be error-prone. Instead, can we just say:
```
pred_probs : np.array (shape (N, K))
Predicted-probabilities in the same format expected by the `score_label_quality()` method.
```
We'd want to use this same brief explanation in docstring of each the individual scoring functions. Then `score_label_quality()` can host the 'master' documentation for the `pred_probs` arg with all the gory details in one place.

Reply:
Agree, done: 100783d9199110a66737f2d09e2fce762b500c45

Thanks